fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -586,15 +589,52 @@ public Response createObject(@QueryParam("checksum") final String checksumDeprec
         }
 
         LOGGER.debug("Finished creating resource with path: {}", newObjectPath);
+        return createUpdateResponse(resource, true);
+    }
 
-        addCacheControlHeaders(servletResponse, result, session);
-
-        final URI location = getUri(result);
+    private Response createUpdateResponse(final FedoraResource resource, final boolean created) throws IOException {
```

Comment:
There is a fair amount of logic in this method. Please add a method comment giving an overview of what is expected in the returned `Response`.
...Or if you prefer, add inline comments throughout the logic of this method.